api_app/analyzers_manager/controller.py

```diff
@@ -253,3 +240,19 @@ def run_analyzer(job_id: int, config: AnalyzerConfig, **kwargs) -> AnalyzerRepor
         report = set_failed_analyzer(job_id, config.name, str(e))
 
     return report
+
+
+def kill_ongoing_analysis(job: Job):
+    qs = job.analyzer_reports.all()
+    # kill celery tasks using task ids
+    task_ids = list(qs.values_list("task_id", flat=True))
+    celery_app.control.revoke(task_ids, terminate=True)
+
+    # update report statuses
+    statuses_to_filter = [
+        AnalyzerReport.Statuses.PENDING.name,
+        AnalyzerReport.Statuses.RUNNING.name,
+    ]
+    qs.filter(status__in=statuses_to_filter).update(
```

Comment:
This `.filter` should be applied before calling `.revoke`. No point in revoking task IDs for analyzers that already finished.